packages/backend/src/models/ContentModel/ContentConfigurations/DashboardContentConfiguration.ts

```diff
@@ -0,0 +1,148 @@
+import { PinnedDashboardTableName } from '../../../database/entities/pinnedList';
+import { ProjectTableName } from '../../../database/entities/projects';
+import { SpaceTableName } from '../../../database/entities/spaces';
+import {
+    ContentConfiguration,
+    ContentFilters,
+    SummaryContentRow,
+} from '../ContentModelTypes';
+
+type SelectSavedSql = SummaryContentRow<{
+    source: 'sql';
+    chart_kind: ChartKind;
+    dashboard_uuid: string | null;
+    dashboard_name: string | null;
+}>;
+
+export const dashboardContentConfiguration: ContentConfiguration<SelectSavedSql> =
+    {
+        shouldQueryBeIncluded: (filters: ContentFilters) =>
+            !filters.contentTypes ||
+            filters.contentTypes?.includes(ContentType.DASHBOARD),
+        getSummaryQuery: (
+            knex: Knex,
+            filters: ContentFilters,
+        ): Knex.QueryBuilder =>
+            knex
+                .from(DashboardsTableName)
+                .innerJoin(
+                    SpaceTableName,
+                    `${SpaceTableName}.space_id`,
+                    `${DashboardsTableName}.space_id`,
+                )
+                .innerJoin(
+                    ProjectTableName,
+                    `${SpaceTableName}.project_id`,
+                    `${ProjectTableName}.project_id`,
+                )
+                .innerJoin(
+                    OrganizationTableName,
+                    `${OrganizationTableName}.organization_id`,
+                    `${ProjectTableName}.organization_id`,
+                )
+                .leftJoin(
+                    PinnedDashboardTableName,
+                    `${PinnedDashboardTableName}.dashboard_uuid`,
+                    `${DashboardsTableName}.dashboard_uuid`,
+                )
+                .select<SelectSavedSql[]>([
+                    knex.raw(`'dashboard' as content_type`),
+                    knex.raw(
+                        `${DashboardsTableName}.dashboard_uuid::text as uuid`,
+                    ),
+                    `${DashboardsTableName}.name`,
+                    `${DashboardsTableName}.description`,
+                    `${DashboardsTableName}.slug`,
+                    `${SpaceTableName}.space_uuid`,
+                    `${SpaceTableName}.name as space_name`,
+                    `${ProjectTableName}.project_uuid`,
+                    `${ProjectTableName}.name as project_name`,
+                    `${OrganizationTableName}.organization_uuid`,
+                    `${OrganizationTableName}.organization_name`,
+                    `${PinnedDashboardTableName}.pinned_list_uuid as pinned_list_uuid`, // TODO: Implement pinned lists
+                    knex.raw(
+                        `${DashboardsTableName}.created_at::timestamp as created_at`,
+                    ),
+                    knex.raw(`NULL as created_by_user_uuid`),
+                    knex.raw(`NULL as created_by_user_first_name`),
```

Comment:
this will be needed on the current dashboard list view

Reply:
Correct, I left a note in the description that some content information is missing. Some content tables don't have the information required without joining the version tables.
We need to standardize the content tables to have all the data we need for this union query. 
That means adding columns like the ones we have for charts `last_version_created_by: uuid` and `last_version_created_at: Date`

I can spend the next hours trying to join the version tables without causing the query to be slow.

Reply:
Here is the SQL joining version tables to find the first and the last version.

```
select 'dashboard'                           as content_type,
       dashboards.dashboard_uuid::text       as uuid,
       "dashboards"."name",
       "dashboards"."description",
       "dashboards"."slug",
       "spaces"."space_uuid",
       "spaces"."name"                       as "space_name",
       "projects"."project_uuid",
       "projects"."name"                     as "project_name",
       "organizations"."organization_uuid",
       "organizations"."organization_name",
       "pinned_dashboard"."pinned_list_uuid" as "pinned_list_uuid",
       first_version.created_at::timestamp   as created_at,
       created_by_user.user_uuid             as created_by_user_uuid,
       created_by_user.first_name            as created_by_user_first_name,
       created_by_user.last_name             as created_by_user_last_name,
       last_version.created_at               as last_updated_at,
       updated_by_user.user_uuid             as last_updated_by_user_uuid,
       updated_by_user.first_name            as last_updated_by_user_first_name,
       updated_by_user.last_name             as last_updated_by_user_last_name,
       "dashboards"."views_count"            as "views",
       dashboards.first_viewed_at::timestamp as first_viewed_at,
       json_build_object()                   as metadata
from "dashboards"
         inner join "spaces" on "spaces"."space_id" = "dashboards"."space_id"
         inner join "projects" on "spaces"."project_id" = "projects"."project_id"
         inner join "organizations" on "organizations"."organization_id" = "projects"."organization_id"
         left join "pinned_dashboard"
                   on "pinned_dashboard"."dashboard_uuid" = "dashboards"."dashboard_uuid"
         left join "dashboard_versions" as "last_version" on "last_version"."dashboard_id" = dashboards.dashboard_id
         left join "dashboard_versions" as first_version on last_version.dashboard_id = dashboards.dashboard_id
         left join "users" as created_by_user on created_by_user.user_uuid = first_version.updated_by_user_uuid
         left join "users" as updated_by_user on updated_by_user.user_uuid = last_version.updated_by_user_uuid
WHERE last_version.dashboard_version_id = (select dashboard_version_id
                                           from dashboard_versions
                                           where dashboard_id = dashboards.dashboard_id
                                           order by dashboard_versions.created_at desc
                                           limit 1)
  AND first_version.dashboard_version_id = (select dashboard_version_id
                                            from dashboard_versions
                                            where dashboard_id = dashboards.dashboard_id
                                            order by dashboard_versions.created_at asc
                                            limit 1)
```